packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -146,7 +146,10 @@ export class ApolloServerBase<
   // The constructor should be universal across all environments. All environment specific behavior should be set by adding or overriding methods
   constructor(config: Config<ContextFunctionParams>) {
     if (!config) throw new Error('ApolloServer requires options.');
-    this.config = config;
+    this.config = {
+      ...config,
+      nodeEnv: 'nodeEnv' in config ? config.nodeEnv : process.env.NODE_ENV,
```

Comment:
This ensures that it stays in line with the old behaviour. Note that as long as the property exists - even if it's `undefined` - it will _not_ default to NODE_ENV.